### PR TITLE
Remove all 'any' types and fix incorrect type definitions

### LINKED ISSUE #24: Remove all 'any' types
We have taken a few shortcuts and set types as 'any' here and there. This is bad practice and should be avoided.

### PR DESCRIPTION
**Pull request by AndreasWintherMoen**
Closes #24 

Fixes backend types (frontend types seem to be good) which were often incorrectly typed, particularly database functions. We should be more strict about this when setting up new routes. Kinda defeats the purpouse of using TypeScript if we don't assign the correct types to our functions. 

Also adds explicit types to all routes by extending Express's Request and Response types. This was a bit of work to set up but I think it should be quite easy to define new types now. Just look at types/routerTypes for inspiration. 

I've also reorganized the backend types into different files like we did previously in the frontend.

### CODE REVIEW

File: backend/.eslintrc.js [hunk 1]
```diff
@@ -18,5 +18,6 @@ module.exports = {
   rules: {
     'prettier/prettier': 'error',
     'no-console': 'off',
+    'import/prefer-default-export': 'off',
```
Comment (erikskaar) [inline backend/.eslintrc.js:21]: Should this be disabled for the entire project and not individual files?

File: backend/src/routes/ontologies.ts [hunk 2]
```diff
@@ -84,26 +94,26 @@ const getSubGoalsfromSDG = async (req, res) => {
...
 
-const regexSearch = async (req, res) => {
+const regexSearch = async (req: RegexRequest, res: NodeArrayResponse) => {
   try {
     const searchTerm = req.query.search;
     const limitResults = req.query.limit;
-    const data = await getClassesByString(searchTerm, limitResults);
-
+    verifyRequestQueryParams(searchTerm);
+    const data = await getClassesByString(searchTerm!, limitResults);
```
Comment (erikskaar) [inline backend/src/routes/ontologies.ts:102]: Can `searchTerm` be guaranteed here or should it be null-checked first?

**Review by erikskaar — CHANGES_REQUESTED** [review]
Look good. Just have a few questions I've noted in the review :)

File: backend/src/routes/ontologies.ts (hunk 2 above)
Comment (AndreasWintherMoen) [reply backend/src/routes/ontologies.ts:102]: verifyRequestQueryParams checks for null values and throws an exception if null or undefined.

File: backend/.eslintrc.js (hunk 1 above)
Comment (AndreasWintherMoen) [reply backend/.eslintrc.js:21]: Yea I guess you're right. I'll fix it.

**Review by erikskaar — APPROVED** [review]
GJ

**Pull request merged**